### PR TITLE
New data source - `cloudflare_origin_ca_root_certificate`

### LINKED ISSUE #1096: New data resource - Origin CA certificate chain
### Current Terraform and Cloudflare provider version

Terraform v0.14.11
+ provider registry.terraform.io/cloudflare/cloudflare v2.21.0

### Description

Add new data resource `cloudflare_origin_ca_certificate_chain`

### Use cases

When applying certificates generated using the resource `cloudflare_origin_ca_certificate` one often needs access to the Cloudflare Origin CA Certificate Chain too. Until now I have been reading this from a link I found via the[ Cloudflare support site](https://developers.cloudflare.com/ssl/origin-configuration/origin-ca#4-required-for-some-add-cloudflare-origin-ca-root-certificates). Unfortunately the link I was using - https://support.cloudflare.com/hc/article_attachments/360037885371/origin_ca_rsa_root.pem - now returns a `404`. The [replacement](https://developers.cloudflare.com/ssl/e2b9968022bf23b071d95229b5678452/origin_ca_rsa_root.pem) does not like programmatic access; trying to access URL this via Terraform/curl returns a `403` with a Cloudflare Captcha page. So, I'm left searching for alternatives... It would be great if this provider handled that complexity for me.

### Potential Terraform configuration

```hcl
data "cloudflare_origin_ca_certificate_chain" "cloudflare" {}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by onematchfox**
Introduces new data source `cloudflare_origin_ca_root_certificate` which can be used to read the Origin CA root certificate for a given algorithm. 

Closes #1096

### CODE REVIEW

File: cloudflare/data_source_origin_ca_root_certificate.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+			},
+		},
+	}
+}
+
+func dataSourceCloudflareOriginCARootCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	algorithm := strings.ToLower(fmt.Sprintf("%v", d.Get("algorithm")))
+	certBytes, err := cloudflare.OriginCARootCertificate(algorithm)
+	if err != nil {
+		return fmt.Errorf("failed to fetch Cloudflare IP ranges: %s", err)
```
Comment (jacobbednarz) [inline cloudflare/data_source_origin_ca_root_certificate.go:36]: copy paste?

```suggestion
		return fmt.Errorf("failed to fetch Cloudflare Origin CA root %s certificate: %s", algorithm, err)
```

File: cloudflare/data_source_origin_ca_root_certificate.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+			"cert_pem": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceCloudflareOriginCARootCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	algorithm := strings.ToLower(fmt.Sprintf("%v", d.Get("algorithm")))
```
Comment (jacobbednarz) [inline cloudflare/data_source_origin_ca_root_certificate.go:33]: as we're using `strings.ToLower`, this should be explicit

```suggestion
	algorithm := strings.ToLower(fmt.Sprintf("%s", d.Get("algorithm")))
```

File: website/docs/d/origin_ca_root_certificate.html.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+layout: "cloudflare"
+page_title: "Cloudflare: cloudflare_origin_ca_root_certificate"
+sidebar_current: "docs-cloudflare-datasource-origin_ca_root_certificate"
```
Comment (jacobbednarz) [inline website/docs/d/origin_ca_root_certificate.html.md:4]:
```suggestion
sidebar_current: "docs-cloudflare-datasource-origin-ca-root-certificate"
```

File: website/cloudflare.erb [hunk 4]
```diff
@@ -28,6 +28,9 @@
             <li<%= sidebar_current("docs-cloudflare-datasource-ip_ranges") %>>
               <a href="/docs/providers/cloudflare/d/ip_ranges.html">cloudflare_ip_ranges</a>
             </li>
+            <li<%= sidebar_current("docs-cloudflare-datasource-origin_ca_root_certificate") %>>
```
Comment (jacobbednarz) [inline website/cloudflare.erb:31]:
```suggestion
            <li<%= sidebar_current("docs-cloudflare-datasource-origin-ca-root-certificate") %>>
```

**Comment by jacobbednarz** [comment]
some minor nitpicks but nothing major 👍 once we get those sorted, this looks fine to merge in.

**Comment by jacobbednarz** [comment]
integration tests look good

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareOriginCARootCertificate" -count 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareOriginCARootCertificate
--- PASS: TestAccCloudflareOriginCARootCertificate (0.76s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/cloudflare	1.138s
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
```

File: cloudflare/data_source_origin_ca_root_certificate.go (hunk 1 above)
Comment (onematchfox) [reply cloudflare/data_source_origin_ca_root_certificate.go]: Oops :roll_eyes:

File: cloudflare/data_source_origin_ca_root_certificate.go (hunk 2 above)
Comment (onematchfox) [reply cloudflare/data_source_origin_ca_root_certificate.go:33]: Good catch

**Comment by onematchfox** [comment]
> some minor nitpicks but nothing major  once we get those sorted, this looks fine to merge in.

Thanks for the review. Accepted all suggestions. Hope you don't mind me tagging an extra commit on to the PR - I took the liberty of fixing the `ip_range` sidebar _annotations_ so that these are no longer the odd ones out with their use of `_` instead of `-`

**Comment by onematchfox** [comment]
Hi @jacobbednarz, Just checking in on this PR to make sure it hasn't dropped off your radar. Let me know if there's anything else you need me to do. Cheers.

**Comment by jacobbednarz** [comment]
this definitely hasn't fallen off my radar; just a little swamped this
week coming back from vacation :)

will hopefully wrap this up this week at some point.

**Comment by jacobbednarz** [comment]
🥳 thanks for this!

**Comment by nboaldin** [comment]
`cloudflare_origin_ca_root_certificate` is returning html code for me, not an actual cert. Wanted to poke in here before I open up an issue.

**Comment by onematchfox** [comment]
> `cloudflare_origin_ca_root_certificate` is returning html code for me, not an actual cert. Wanted to poke in here before I open up an issue.

Yeah, unfortunately Cloudflare adjusted the URLs that were being used to retrieve the certs. This has been fixed in https://github.com/cloudflare/cloudflare-go/pull/819 but we'll need to wait for a new release of that library to be able to fix this one.

cc @jacobbednarz

**Pull request merged**